OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -357,6 +357,33 @@ void IActivityNotifyStanceChanged.StanceChanged(Actor self, AutoTarget autoTarge
 				if (!autoTarget.HasValidTargetPriority(self, lastVisibleOwner, lastVisibleTargetTypes))
 					attack.RequestedTarget = Target.Invalid;
 			}
+
+			void IActivityNotifyPreemptiveCancel.CancelledBeforeFirstTick(Actor self)
+			{
+				// AttackFollow.OnQueueAttackActivity may have preemptively set attack.RequestedTarget
+				// We will not have a chance to take over control for this target, so must reset
+				// the target to prevent the OpportunityFire logic from being blocked
+				if (target.Type != attack.RequestedTarget.Type || target.Type == TargetType.Invalid)
+					return;
+
+				if (target.Type == TargetType.Actor && target.Actor != attack.RequestedTarget.Actor)
```

Comment:
These checks will fail if the target transitions between actor/frozenactor between assignment and cancellation.

Either the target check needs to be smart enough to consider FrozenActor/Actor equivalency, or we can store and check against the activity directly instead of the target.